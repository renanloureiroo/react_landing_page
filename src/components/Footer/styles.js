import styled, { css } from 'styled-components';

import { TextComponent } from '../TextComponent/index';
export const Container = styled.div`
  ${({ theme }) => css`
    & a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.sizes.small};
    }
    text-align: center;
  `}
`;

export const TextFooter = styled(TextComponent).attrs('div')``;
