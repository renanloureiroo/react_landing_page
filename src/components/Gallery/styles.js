import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css``}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.huge};
  `}
`;
export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    justify-content: center;
    gap: ${theme.spacing.medium};

    & div {
      width: 100%;
      height: 100%;
      margin: 0 auto;
      overflow: hidden;
    }

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.img`
  transition: all 300ms ease-in-out;
  ${() => css`
    width: 100%;
    height: 100%;

    &:hover {
      transform: scale(1.2) rotate(7deg);
    }
  `}
`;
