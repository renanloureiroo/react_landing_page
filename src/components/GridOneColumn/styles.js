import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    max-width: 50%;
    margin: 0 auto;
    text-align: center;
  `}
`;
