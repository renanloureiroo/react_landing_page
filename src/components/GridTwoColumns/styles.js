import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1.5fr;
    gap: ${theme.sizes.small};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    } ;
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;

    & h1 {
      margin-bottom: ${theme.sizes.small};
    }
  `}
`;
export const ImgContainer = styled.div`
  ${() => css`
    width: 100%;
    height: 100%;
    margin: 0 auto;

    & img {
      width: 100%;
      height: 100%;
    }
  `}
`;
