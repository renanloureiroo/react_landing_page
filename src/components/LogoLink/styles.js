import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${() => css`
    display: flex;
    text-decoration: none;
    color: inherit;

    & > img {
      max-height: 3rem;
    }
  `}
`;
