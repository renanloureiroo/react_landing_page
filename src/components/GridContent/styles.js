import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    margin: 0 auto;
    max-width: 58rem;
    grid-template-columns: 1fr;
    text-align: center;
    justify-content: center;
    ${Title} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
  & p{
    margin-bottom: ${theme.spacing.large}};
  }
  `}
`;
