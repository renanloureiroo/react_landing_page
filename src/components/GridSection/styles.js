import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-rows: 1fr 2fr;
    gap: ${theme.spacing.large};

    ${Title} {
      margin-bottom: ${theme.spacing.small};
    }

    @media ${theme.media.lteMedium} {
      display: block;
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.large};
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: ${theme.spacing.medium};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    max-width: 36rem;
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.spacing.xhuge};
      font-weight: bold;
      transform: rotate(10deg);
      left: -5rem;
      top: -2rem;
    }
  `}
`;
