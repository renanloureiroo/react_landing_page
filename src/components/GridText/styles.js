import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-rows: 1fr 2fr;
    gap: ${theme.spacing.large};

    ${Title} {
      margin-bottom: ${theme.spacing.small};
    }

    @media ${theme.media.lteMedium} {
      display: block;
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.large};
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacing.large};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.medium};
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.spacing.xhuge};
      font-weight: bold;
      transform: rotate(10deg);
      left: -5rem;
      top: -2rem;
    }
  `}
`;
