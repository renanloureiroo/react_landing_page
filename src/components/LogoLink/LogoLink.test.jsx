import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { LogoLink } from '.';

describe('<LogoLink/>', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink text="Logo" link="#target" />);
    const heading = screen.getByRole('heading', { name: 'Logo' });
    expect(heading.firstChild).toHaveAttribute('href', '#target');
  });
  it('should render image logo', () => {
    renderTheme(<LogoLink text="Logo" srcImg="image.jpg" link="#target" />);
    screen.getByRole('heading', { name: 'Logo' });
    expect(screen.getByRole('img', { name: 'Logo' })).toHaveAttribute(
      'src',
      'image.jpg',
    );
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink text="Logo" srcImg="image.jpg" link="#target" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
