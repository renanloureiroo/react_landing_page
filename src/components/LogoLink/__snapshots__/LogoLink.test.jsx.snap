// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink/> should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1 > img {
  max-height: 3rem;
}

.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4rem;
  }
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="Logo"
      src="image.jpg"
    />
  </a>
</h1>
`;
